Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1540,17 +1540,46 @@ extension BrowserViewController: URLBarDelegate {
     }
 
     func urlBar(urlBar: URLBarView, didSubmitText text: String) {
-        // If we can't make a valid URL, do a search query.
-        // If we still don't have a valid URL, something is broken. Give up.
-        let engine = profile.searchEngines.defaultEngine
-        guard let url = URIFixup.getURL(text) ??
-                        engine.searchURLForQuery(text) else {
-            log.error("Error handling URL entry: \"\(text)\".")
+        let trimmedText = text.stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceCharacterSet())
+        
+        guard trimmedText.contains(" ") else {
+            submitText(text)
             return
         }
 
-        Telemetry.recordEvent(SearchTelemetry.makeEvent(engine: engine, source: .URLBar))
+        let possibleKeywordQuerySeparatorSpace = trimmedText.characters.indexOf(" ")!
+        let possibleKeyword = trimmedText.substringToIndex(possibleKeywordQuerySeparatorSpace)
+        let possibleQuery = trimmedText.substringFromIndex(possibleKeywordQuerySeparatorSpace.successor())
+        
+        profile.bookmarks.getURLForKeywordSearch(possibleKeyword).uponQueue(dispatch_get_main_queue()) {
+                if let urlString = $0.successValue {
```

Comment:
@thebnich, I was told that Mozilla devs use a linting tool for swift. Which one is it? There seem to be quite a few such tools available. I tried [SwiftLint](https://github.com/realm/SwiftLint), but it seems to be running only a [small subset of the rules](https://github.com/realm/SwiftLint#rules). Also, these rules are based on [Github's style guide](https://github.com/github/swift-style-guide) and firefox uses a [different one](https://github.com/mozilla-mobile/firefox-ios#swift-style).